src/engine/input/key-mapping/InputGroupList.cs

```diff
@@ -0,0 +1,180 @@
+
+        // Take all InputGroups.
+        // Take the ones with any interception of the environments.
+        // Take the input actions.
+        // Get the first action where the event collides or null if there aren't any.
+        return AllGroupItems.Where(p => p.Environments.Any(x => environments.Contains(x)))
+            .SelectMany(p => p.Actions)
+            .Where(p => !Equals(item.AssociatedAction, p))
+            .SelectMany(p => p.Inputs)
+            .FirstOrDefault(p => Equals(p, item));
+    }
+
+    public void InitFromData(Dictionary<string, List<InputEventWithModifiers>> data)
+    {
+        foreach (Node child in GetChildren())
+        {
+            RemoveChild(child);
+        }
+
+        var inputGroupItemScene = GD.Load<PackedScene>("res://src/engine/input/key-mapping/InputGroupItem.tscn");
+        var inputActionItemScene = GD.Load<PackedScene>("res://src/engine/input/key-mapping/InputActionItem.tscn");
+
+        // Load json
+        var definition =
+            new[]
+            {
+                new
+                {
+                    EnvId = new[]
+                    {
+                        string.Empty,
+                    },
+                    GroupName = string.Empty,
+                    Actions = new[]
+                    {
+                        new
+                        {
+                            InputName = string.Empty,
+                            DisplayName = string.Empty,
+                        },
+                    },
+                },
+            };
+
+        using var file = new File();
+        file.Open("res://simulation_parameters/common/input_options.json", File.ModeFlags.Read);
+        var fileContent = file.GetAsText();
+        var jsonResult = JsonConvert.DeserializeAnonymousType(fileContent, definition);
+        file.Close();
+
+        // Add inputGroups to the DOM
+        foreach (var groupItem in jsonResult)
+        {
+            // Create InputGroupItem
+            var input = (InputGroupItem)inputGroupItemScene.Instance();
+            input.Actions = groupItem.Actions.Select(p =>
+            {
+                var inputEvents = data[p.InputName];
+
+                // Create InputActionItem
+                var inputAction = (InputActionItem)inputActionItemScene.Instance();
+                inputAction.InputName = p.InputName;
+                inputAction.DisplayName = p.DisplayName;
+                inputAction.Inputs = new ObservableCollection<InputEventItem>(
+                    inputEvents.Select(x =>
+                    {
+                        var inputEvent = (InputEventItem)InputEventItemScene.Instance();
+                        inputEvent.AssociatedEvent = x;
+                        inputEvent.AssociatedAction = inputAction;
+                        return inputEvent;
+                    }));
+                inputAction.AssociatedGroup = input;
+                return inputAction;
```

Comment:
I think this code should be flattened, to me it doesn't look good to have this much code in a Select.
Using LINQ isn't applicable everywhere.